File: 66. 加一/66. 加一.py
class Solution:
    def plusOne(self, digits):
        jinwei = 1
        n = len(digits)
        for i in range(n):
            if(jinwei!=1):
                break
            digits[n-i-1] += 1
            if(digits[n-i-1] == 10):
                jinwei = 1
                digits[n-i-1] = 0
            else:
                return digits
        #仍存在进位，说明位数+1，数组位数增加
        return [1] + [0 for i in range(n)]
s = Solution()
print(s.plusOne([9,9,6]))

File: 66. 加一/66. 加一.go
package main

import "fmt"
// 作为go语言学习，按照python版翻译得来
func plusOne(digits []int) []int {
    jinwei := 1
    n := len(digits)
    for i := n-1; i >= 0; i-- {
        if jinwei!=1 {
            break
        }
        digits[i] ++
        if digits[i] == 10 {
            jinwei = 1
            digits[i] = 0
        }else{
            return digits
        }
    }
    return append([]int{1},digits...)
}

func main()  {
    digits := []int{9,9}
    fmt.Println(plusOne(digits))
}